lib/services.js

```diff
@@ -241,22 +242,39 @@ const services = {
         log.trace('deleting object from bucket');
         assert.strictEqual(typeof bucketName, 'string');
         assert.strictEqual(typeof objectMD, 'object');
-        return metadata.deleteObjectMD(bucketName, objectKey, options, log,
-            (err, res) => {
+
+        function deleteMDandData() {
+            return metadata.deleteObjectMD(bucketName, objectKey, options, log,
+                (err, res) => {
+                    if (err) {
+                        return cb(err, res);
+                    }
+                    cb(null, res); // this is smart
+                    log.trace('deleteObject: metadata delete OK');
+                    const deleteLog = logger.newRequestLogger();
+                    if (objectMD.location === null) {
+                        return undefined;
+                    } else if (!Array.isArray(objectMD.location)) {
+                        return data.delete(objectMD.location, deleteLog);
+                    }
+                    return data.batchDelete(objectMD.location, null, null,
+                    deleteLog);
+                });
+        }
+
+        const objGetInfo = objectMD.location;
+        if (objGetInfo && objGetInfo[0].dataStoreType === 'azure') {
+            return externalBackendUtils.protectAzureBlocks(bucketName,
```

Comment:
is there a way to keep the protection code in the azure client so we don't have to spread the logic around?

Reply:
the function is in a util file in the external folder, and I put it there cos we don't really call any of the external clients without going through the gateway, so I thought that was a good compromise. I can move it if you'd still prefer

Reply:
I meant, when an action is called in the azure client itself, can we put the protection check there?  that way, anyone extending will not have to worry about calling protectAzureBlocks outside of the azure client.